docs/proposals/images/machinepool-machines/machinepool-machines-reconcile.plantuml

```diff
@@ -0,0 +1,23 @@
+@startuml machinepool-machine-reconcile
+:MachinePool reconciliation begins;
+if (status.InfrastructureMachineKind is set on InfraMachinePool) then (no)
+  :reconcile MachinePool replicas without MachinePool Machines;
+else (yes)
+  :find the InfraMachinePoolMachines matching the kind
+  with the matching MachinePool name and Cluster name label;
+  repeat 
+    :select an InfraMachinePoolMachine;
+    if (a owner Machine exists for a InfraMachinePoolMachine) then (no);
+      :create a Machine owned by the MachinePool with
+      infraRef pointing to the InfraMachinePoolMachine;
+      :Machine controller automatically sets ownerRef
+      of InfraMachinePoolMachine to Machine;
+    else (yes)
+      :ensure Machine has an ownerRef to the MachinePool;
```

Comment:
is this really an else or are we ensuring the owner ref this no matter what?

Reply:
Yeah, I think it's fair to move this out of the "else" branch.